lacommunaute/notification/signals.py

```diff
@@ -0,0 +1,16 @@
+from lacommunaute.notification.enums import EmailSentTrackKind, NotificationDelay
+from lacommunaute.notification.models import Notification
+
+
+# registered in apps.py
+def create_post_notifications(sender, instance, **kwargs):
+    """
+    When a Post is created, we schedule notifications to users active in the thread
+    """
+    delay = NotificationDelay.ASAP if instance.is_first_reply else NotificationDelay.DAY
+
+    notifications = [
+        Notification(recipient=email_address, post=instance, kind=EmailSentTrackKind.NEW_MESSAGES, delay=delay)
+        for email_address in instance.topic.mails_to_notify()
```

Comment:
note to self : verifier si `mails_to_notify` inclue les utilisateurs qui ont sauvegardé le `forum` du `topic`

Reply:
Normalement non

```
def mails_to_notify(self:
    # we want to notify stakeholders of the topic, except the last poster.
    # stakeholders are:
    # - authenticated users who upvoted one of the posts of the topic
    # - authenticated users who posted in the topic
    # - anonymous users who posted in the topic

    authenticated_qs = (
        User.objects.filter(
                Q(
                    upvotes__content_type=ContentType.objects.get_for_model(Post),
                    upvotes__object_id__in=self.posts.values("id"),
                )
                | Q(posts__topic=self)
        )
    )

    # ...
```

Reply:
je l'ajouterai dans une PR à venir